articles/build-an-outlook-clone-using-react-hooks/index.md

```diff
@@ -0,0 +1,881 @@
+    }
+  ]
+}
+
+```
+
+### Initiate JSON Server
+```
+json-server --watch db.json
+```
+
+If you get an error, do check if **npm** has been added to the environment variables. That should be a quick fix. Add the following to your **PATH** environment variable: C:\Users\YOUR_NAME\AppData\Roaming\npm
+
+Once we have the server running, we can check via Postman or just go to http://localhost:3000/emailbody. To get a specific email body, we can send the *id* as a parameter. The URL is modified as follows: http://localhost:3000/emailbody?id=1. This returns the first email body.
+
+### Let's implement the hooks we just learned
+
+In the earlier article, we have discussed the logic to go about implementing the project from scratch. In this article, we will give more importance to understanding the hooks and performing optimization.
+
+### Explaination: *App.js*
+
+We declare functional components and hooks need to be declared within the function. We use various state variables. They are
+1. emails
+2. dualPanel
+3. id
+4. body
+5. searchField
+6. readList
+7. fav
+8. forRead
+9. forSearch
+10. forFav
+11. unread
+
+These are the same state variables as earlier. This time they are declared as and when used. This is a powerful feature. Declare state variables as and when required.
+
+Initially, on componentDidMount, the first useEffect is called, where we fetch the list. Initially, the id is set to an empty string. When there is a change in id, we will make an API call to the localhost to fetch the body. The body state is set. getId is the callback function that is passed as a prop to Email component and we set the id.
+
+Earlier, we had to send a callback function to modify the readlist. In this case, we can directly send the useState list. We send readlist and setReadList as props to *Email.js*. In *Email.js*, we setReadList on an onClick event.
+
+We have also used createContext. We declare a global const EmailContext. This is wrapped around the child component. We send the emails as the value to be received at the child component. In the child component, we use the hook useContext to receive the prop email.
+
+### Styled-components
+
+This is by far the easiest library to use for styling. There are various alternatives like inline styling and tachyons. But styled-components enables us to use our CSS knowledge. Let's say we want to style a button element. The inline styling would be
```

Comment:
This is by far the easiest library to use for styling.
While I understand you are referencing the heading of this section when you say "This", it is not correct to do so. You should replace "This" with "Styled-components".